src/appleseedmaya/exporters/shadingnodeexporter.cpp

```diff
@@ -58,6 +58,12 @@ namespace asr = renderer;
 namespace
 {
 
+const char *g_compToColorParamNames[] = {"compR", "compG", "compB"};
+const char *g_compToColorOutputParamName = "comp";
+
+const char *g_colorToCompParamNames[] = {"compR", "compG", "compB"};
```

Comment:
I'm not sure i follow. The suffix R,G,B or X,Y,Z or U,V are added by maya for compound attributes. The only reason these would change would be to reflect some upstream change in Maya, but even they would remain the same until we changed them in the shaders for the type2component and component2type shaders that split the compound types. The names are set presently [here](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.shaders/src/maya/as_maya_color2Components.osl) and [here](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.shaders/src/maya/as_maya_components2Color.osl) for instance.